src/Shared/CustomPagination/CustomPagination.tsx

```diff
@@ -1,81 +1,110 @@
-import React, { useState } from 'react';
-import { KeyboardEvent } from 'react';
+import React, { useState } from "react";
+import { KeyboardEvent } from "react";
 interface CustomPaginationProps {
-    totalPages: number;
-    currentPage: number;
-    onPageChange: (page: number) => void;
+  totalPages: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
 }
 
+const CustomPagination: React.FC<CustomPaginationProps> = ({
+  totalPages,
+  currentPage,
+  onPageChange,
+}) => {
+  const visiblePageLimit = 5; // Number of visible pages
+  const halfVisiblePageLimit = Math.floor(visiblePageLimit / 2);
 
-const CustomPagination: React.FC<CustomPaginationProps> = ({ totalPages, currentPage, onPageChange }) => {
-    const visiblePageLimit = 5; // Number of visible pages
-    const halfVisiblePageLimit = Math.floor(visiblePageLimit / 2);
+  const startPage = Math.max(1, currentPage - halfVisiblePageLimit);
+  const endPage = Math.min(totalPages, startPage + visiblePageLimit - 1);
 
-    const startPage = Math.max(1, currentPage - halfVisiblePageLimit);
-    const endPage = Math.min(totalPages, startPage + visiblePageLimit - 1);
+  const pages = Array.from(
+    { length: endPage - startPage + 1 },
+    (_, index) => startPage + index
+  );
 
-    const pages = Array.from({ length: endPage - startPage + 1 }, (_, index) => startPage + index);
+  const [pageNumberInput, setPageNumberInput] = useState("");
 
-    const [pageNumberInput, setPageNumberInput] = useState('');
+  const handlePageNumberInputChange = (
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setPageNumberInput(e.target.value);
+  };
 
-    const handlePageNumberInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-        setPageNumberInput(e.target.value);
-    };
+  const handleGoToPage = () => {
+    const pageNumber = parseInt(pageNumberInput);
+    if (pageNumber >= 1 && pageNumber <= totalPages) {
+      onPageChange(pageNumber);
+    }
+  };
+  const handlePageNumberInputKeyPress = (
+    e: KeyboardEvent<HTMLInputElement>
+  ) => {
+    if (e.key === "Enter") {
+      handleGoToPage();
+    }
+  };
 
-    const handleGoToPage = () => {
-        const pageNumber = parseInt(pageNumberInput);
-        if (pageNumber >= 1 && pageNumber <= totalPages) {
-            onPageChange(pageNumber);
-        }
+  return (
+    <nav aria-label="..." className="">
+      <ul className=" m-0 pagination justify-content-center pagination-sm">
+        {currentPage > 1 && (
+          <button className="paginate btn-hover-custom bg-white p-0 m-0 border-0" onClick={() => onPageChange(currentPage - 1)}>
+            <li className="page-item">
+              <a className="page-link">Prev</a>
+            </li>
+          </button>
+        )}
 
-    };
-    const handlePageNumberInputKeyPress = (e: KeyboardEvent<HTMLInputElement>) => {
-        if (e.key === 'Enter') {
-            handleGoToPage();
-        }
-    };
+        {pages.map((page) => (
```

Comment:
There are a couple of issues with the code here:

1. The `<button>` element should not contain a `<li>` element directly. Instead, you can apply the classes directly to the `<button>` element.
2. The `key` attribute should be placed on the outermost JSX element inside the `map` function. In this case, it should be on the `<button>` element.
3. The `<a>` element is not allowed directly inside a `<button>` element, it would also be fine if you put the text directly in the button itself.